api/types.go

```diff
@@ -34,9 +34,9 @@ type sandboxResource struct {
 
 // networkCreate is the expected body of the "create network" http request message
 type networkCreate struct {
-	Name        string   `json:"name"`
-	NetworkType string   `json:"network_type"`
-	Labels      []string `json:"labels"`
+	Name        string            `json:"name"`
+	NetworkType string            `json:"network_type"`
+	Labels      map[string]string `json:"labels"`
```

Comment:
`Labels` is optional. I noticed we do not nil check before calling `libnetwork.NetworkOptionLabels(create.Labels)` .
(Check should be done in api.go, not in the option setter itself.)
Should we also add the `json:",omitempty"` tag ?